lib/sol/SolRpc.js

```diff
@@ -1,322 +1,454 @@
-const Web3 = require('@solana/web3.js');
+const {
+  address,
+  createSolanaRpc,
+  isAddress,
+  createTransactionMessage,
+  setTransactionMessageLifetimeUsingBlockhash,
+  getComputeUnitEstimateForTransactionMessageFactory,
+  getTransactionDecoder,
+  getSignatureFromTransaction,
+  sendAndConfirmTransactionFactory,
+  createSolanaRpcSubscriptions,
+  setTransactionMessageFeePayerSigner,
+  appendTransactionMessageInstructions,
+  appendTransactionMessageInstruction,
+  setTransactionMessageLifetimeUsingDurableNonce,
+  sendAndConfirmDurableNonceTransactionFactory,
+  signTransactionMessageWithSigners,
+
+  signature,
+  generateKeyPairSigner,
+  getBase64EncodedWireTransaction,
+} = require('@solana/web3.js');
+const {
+  getSetComputeUnitLimitInstruction,
+  getSetComputeUnitPriceInstruction,
+} = require('@solana-program/compute-budget');
+const { getTransferSolInstruction, getCreateAccountInstruction, SYSTEM_PROGRAM_ADDRESS, getInitializeNonceAccountInstruction } = require('@solana-program/system');
+
+const { pipe } = require('@solana/functional');
 const EventEmitter = require('events');
 const bs58 = require('bs58');
+
 class SolRPC {
 
   /**
-   * Constructs a new instance of the SolRPC class.
-   * 
-   * @param {Object} config - The configuration object containing the connection details.
-   * @param {string} config.protocol - The network protocol (e.g., 'http', 'https', 'wss').
-   * @param {string} config.host - The host URL or IP address.
-   * @param {number} [config.port] - The port number (optional).
-   * @param {string} config.account - The account to use for transactions.
-   */
+ * Constructs a new instance of the SolRPC class.
+ * 
+ * @param {Object} config - The configuration object containing the connection details.
+ * @param {string} config.protocol - The network protocol (e.g., 'http', 'https', 'wss').
+ * @param {string} config.host - The host URL or IP address.
+ * @param {number} [config.port] - The port number (optional).
+ * @param {string} config.account - The account to use for transactions.
+ */
   constructor(config) {
     this.config = config;
-    this.connection = this.initConnection(this.config);
+    this.rpcUrl = `${this.config.protocol}://${this.config.host}`;
+    this.rpcSubscriptionUrl = this.rpcUrl.replace('http', 'ws');
+    this.rpc = this.initRpcConnection(this.config);
+    this.rpcSubscriptions = this.initRpcSubscriptions(this.config);
```

Comment:
You're passing in `this.config` but not doing anything with it and using `this.rpcUrl` instead